projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -32,6 +32,9 @@
 // Each of these is a class that will register its own routes on 'rest_api_init'.
 require_once JETPACK__PLUGIN_DIR . '_inc/lib/core-api/load-wpcom-endpoints.php';
 
+// Create discount after site has been successfully connected.
+add_action( 'jetpack_authorize_ending_authorized', array( 'Jetpack_Core_Json_Api_Endpoints', 'create_site_discount' ) );
```

Comment:
Is this the best place to add that action?

Reply:
We're not actually using `create_site_discount` as an endpoint, right? Instead it's used to make the call to the WPCOM API.  If that's true then it seems like this and the function itself might fit better somewhere else. I'm not familiar enough with the code base to know where though. Maybe somewhere that gets executed after a successful connection is established?

My understanding is that `create_site_discount` would be called after a successful connection and can be called directly rather than via an API request. Let me know if I'm mistaken.

Reply:
Thanks @travisw!

> We're not actually using `create_site_discount` as an endpoint, right? Instead it's used to make the call to the WPCOM API.

That's right, `create_site_discount` is called directly by the hook, and calls the WPCOM API.

> My understanding is that `create_site_discount` would be called after a successful connection and can be called directly rather than via an API request. Let me know if I'm mistaken.

True, that's the current implementation. Maybe I don't understand your statement correctly.

>  I'm not familiar enough with the code base to know where though. Maybe somewhere that gets executed after a successful connection is established?

Gotcha, I'll ask around.

Reply:
I updated the hook for `jetpack_site_registered`, as it was recommended to me. It's more appropriate as the coupon is site based.

Reply:
I'm still wondering if the code for creating a coupon should actually live in a different area (not in `class.core-rest-api-endpoints.php`) since it's not really defining an endpoint, but rather just calling a wpcom endpoint. But maybe Crew can clarify that once they give it a review.

Reply:
As discussed, the endpoint to create the discount was removed from this PR.